### PR TITLE
Fix(maangchi): use regex string to find ingredients header 

### LINKED ISSUE #1310: (Maangchi) Ingredients not being correctly scraped
**Pre-filing checks**

- [x] I have searched for open issues that report the same problem
- [x] I have checked that the bug affects the latest version of the library

**Context**
Certain recipes have the number of people being served contained within the same heading tag as the ingredients string (see links below). This may be the cause of the ingredients for these recipes not being correctly scraped, leading to a NoneType error when calling the scraper's ingredients method

**The URL of the recipe(s) that are not being scraped correctly**

- https://www.maangchi.com/recipe/budae-jjigae
- https://www.maangchi.com/recipe/gimbap
- https://www.maangchi.com/recipe/maeuntang

**The results you expect to see**

List of ingredients for the above recipes

**The results (including any Python error messages) that you are seeing**

```
Traceback (most recent call last):
  File "/path/to/file", line 14, in <module>
    print(scraper.ingredients())
          ^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/exception_handling.py", line 63, in decorated_method_wrapper
    return decorated(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/html_tags_stripper.py", line 74, in decorated_method_wrapper
    decorated_func_result = decorated(self, *args, **kwargs)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/schemaorg_fill.py", line 57, in decorated_method_wrapper
    return decorated(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/anaconda3/lib/python3.11/site-packages/recipe_scrapers/maangchi.py", line 11, in ingredients
    before = self.soup.find("h2", string="Ingredients").find_all_next("li")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'find_all_next'
make: *** [run] Error 1
```

### PR DESCRIPTION
**Pull request by oahnh**
### Resolves #1310 

Utilizes a regex string to find ingredient headers matching patterns like "Ingredients (serves _n_ to _m_)"

Example recipes
- https://www.maangchi.com/recipe/gimbap 
- https://www.maangchi.com/recipe/budae-jjigae
- https://www.maangchi.com/recipe/maeuntang

Unittests passing:
image

Any feedback or discussion is appreciated!

### CODE REVIEW

**Comment by jknndy** [comment]
This seems reasonable to me!

File: recipe_scrapers/maangchi.py [hunk 1]
```diff
@@ -8,7 +10,9 @@ def host(cls):
         return "maangchi.com"
 
     def ingredients(self):
-        before = self.soup.find("h2", string="Ingredients").find_all_next("li")
+        before = self.soup.find(
+            "h2", string=re.compile(r"Ingredients(\s*\(.*\))?")
+        ).find_all_next("li")
```
Comment (jayaddison) [inline recipe_scrapers/maangchi.py:15]: We might be able to simplify this by removing the bracket-matching: `BeautifulSoup`'s [filtering](https://beautiful-soup-4.readthedocs.io/en/latest/#kinds-of-filters) behaves differently for plain strings (`"Ingredients"`) compared to regular expressions (`re.compile(r"Ingredients")`).

(plain strings exact-match on entire element content; regular expressions are searched for the pattern anywhere within the content)

```suggestion
        before = self.soup.find(
            "h2", string=re.compile(r"Ingredients")
        ).find_all_next("li")
```

Comment (oahnh) [reply recipe_scrapers/maangchi.py]: I didn't know this, good to know!

Comment (jayaddison) [reply recipe_scrapers/maangchi.py]: You're welcome!

**Comment by oahnh** [comment]
I realized both of the commits got squashed into the formatting commit 🤦🏽 I updated per your suggestion in the `chore: formatting` commit

**Review by jayaddison — APPROVED** [review]
Looks good to me too - thank you for contributing, @oahnh!

**Pull request merged**